homeassistant/components/zha/core/device.py

```diff
@@ -99,8 +100,13 @@ def __init__(
             self._zigpy_device.__class__.__module__,
             self._zigpy_device.__class__.__name__,
         )
+        self._consider_unavailable_time = (
+            _CONSIDER_UNAVAILABLE_MAINS
+            if self.is_mains_powered
+            else _CONSIDER_UNAVAILABLE_BATTERY
+        )
         keep_alive_interval = random.randint(*_UPDATE_ALIVE_INTERVAL)
-        self._available_check = async_track_time_interval(
+        self._cancel_available_check = async_track_time_interval(
```

Comment:
Lets keep a list of all things we need to cancel on device removal. I'm gonna need another listener in near future, so probably gonna be easier to keep them in the list.

Reply:
There is already one (a list of unsubs) on the device. Will do another PR after this to handle that. There are a few spots that need it. I wasn’t sure if exposing that made sense but I think it does.